### ISSUE TITLE
Cats Effect 3 migration

### ISSUE THREAD

**Issue by kubukoz** [finagle/finch#1319]
Hi! Are there any plans in place to migrate to Cats Effect 3? If there are any blockers, it would be best to catch them before the RCs are being released. Asking for https://github.com/typelevel/cats-effect/issues/1330 :)

**Comment by sergeykolbasov** [comment]
Hello

Yes, however, this project needs some attention to update documentation first. I'll try to come back to it as soon as possible.

**Comment by stephen-lazaro** [comment]
Would a PR that executed this migration be welcomed?

**Comment by tpetillot** [comment]
Proposal: https://github.com/finagle/finch/pull/1472

**Comment by JackTreble** [comment]
a plan for RC1 was outlined [here](https://github.com/finagle/finch/pull/1472#issuecomment-1102741364)
is this still the plan & what steps are still outstanding?

**Comment by joroKr21** [comment]
Yes we need to decide on #1481 and #1478 - I think the former can be completed and merged, I'm not so sure about the latter.

**Comment by tpetillot** [comment]
Indeed, I don't think we should block the release candidate for a spec refactoring https://github.com/finagle/finch/pull/1478.
I may complete https://github.com/finagle/finch/pull/1481 if needed, let me know! 😃

**Comment by joroKr21** [comment]
> I may complete https://github.com/finagle/finch/pull/1481 if needed, let me know! 😃

All yours, I think it's a good idea 👍

**Comment by tpetillot** [comment]
As I don't know if you got notifications on the pull request, I let you know that https://github.com/finagle/finch/pull/1481 have been completed ✔️

**Comment by rosvit** [comment]
Hi folks, I would like to ask if there is a plan to release cats-effect 3 RC in the near future? Or alternatively, can you recommend using current implementation (available in the master branch) in the production code? This is not meant to rush anything, just seeking an advice, since we would like to update our project to cats-effect 3.
Thanks in advance!

**Comment by joroKr21** [comment]
@tpetillot I think we are ready to release but I don't remember what is the procedure.
Is it enough to publish a tag? 🤔

**Comment by rpless** [comment]
@joroKr21 You can push a commit that removes the `-SNAPSHOT` from the version in version.sbt and CI will handle the rest. Something like this should work: https://github.com/finagle/finch/commit/9b18fb743fa62abc1fff74452133a9c93fb22567

**Comment by joroKr21** [comment]
I released 0.34.0 with CE3 support. I didn't release 1.0 because I realised there are a lot of outstanding issues that might be potentially breaking binary compatibility. So I plan to go through them and check what is worth pursuing with fresh eyes.
